unikernel.ml

```diff
@@ -34,11 +34,16 @@ module Main
 
     (* cache control: all resources use last-modified + etag of last commit *)
     let retrieve_last_commit store =
-      Store.digest store Mirage_kv.Key.empty >>= fun last_hash ->
-      Store.last_modified store Mirage_kv.Key.empty >|= fun r ->
-      let v = Result.fold ~ok:Fun.id ~error:(fun _ -> Pclock.now_d_ps ()) r in
-      let last_date = ptime_to_http_date (Ptime.v v) in
-      last := (last_date, Result.get_ok last_hash)
+      Store.digest store Mirage_kv.Key.empty >>= function
+      | Error (`Not_found _) ->
+        let msg = Fmt.str "No commit found - are you on the correct branch?" in
```

Comment:
That's indeed a good way (we probably should print out which branch we want to clone).

Reply:
Which branch are you thinking about? My thinking was that the user knows what branch is passed, and multiple possible branches can be present - so I wouldn't know what branch to suggest

Reply:
Indeed, on Debian, the default branch still is `master` and we currently occur several cases where people don't really understand this kind of failure. I will try to provide an accessor on `git-kv` to get the branch used.